repose-aggregator/functional-tests/spock-functional-test/src/test/groovy/features/filters/clientauthn/ClientAuthFilterAddHeadersTest.groovy

```diff
@@ -0,0 +1,143 @@
+        identityEndpoint = deproxy.addEndpoint(properties.identityPort,
+                'identity service', null, fakeIdentityService.handler)
+
+
+    }
+
+    def cleanupSpec() {
+        if (deproxy)
+            deproxy.shutdown()
+        if (repose)
+            repose.stop()
+    }
+
+    def setup() {
+        fakeIdentityService.resetHandlers()
+        fakeIdentityService.resetDefaultParameters()
+    }
+
+    // REP-2464: Auth filter should add headers not replace headers
+    // Note: V2 x-pp-user using username
+    def "Verify V2 Auth filter should add headers instead of replace headers"() {
+        given:
+        fakeIdentityService.with {
+            client_token = UUID.randomUUID().toString()
+            tokenExpiresAt = (new DateTime()).plusDays(1);
+            client_tenant = "12345"
+            client_tenant_file = "nast-id"
+            service_admin_role = "not-admin"
+        }
+
+        when:
+        "User passes a request through repose with tenant"
+        MessageChain mc = deproxy.makeRequest(
+                url: "$reposeEndpoint/servers/12345",
+                method: 'GET',
+                headers: [
+                        'content-type': 'application/json',
+                        'X-Auth-Token': fakeIdentityService.client_token,
+                        'x-tenant-id' : 'repose-add-tenant',
+                        'x-pp-groups' : 'repose-test-add-group',
+                        'x-pp-user'   : 'repose-test-add-user',
+                        'x-roles'  : 'test-add-role'])
+
+        then: "Everything gets passed as is to the origin service (no matter the user)"
+        mc.receivedResponse.code == "200"
+        mc.handlings.size() == 1
+        (mc.handlings[0].request.headers.findAll("x-tenant-id").toString()).contains("12345")
+        (mc.handlings[0].request.headers.findAll("x-tenant-id").toString()).contains("nast-id")
+        (mc.handlings[0].request.headers.findAll("x-tenant-id").toString()).contains("repose-add-tenant")
+        (mc.handlings[0].request.headers.findAll("x-pp-groups").toString()).contains("0")
+        (mc.handlings[0].request.headers.findAll("x-pp-groups").toString()).contains("repose-test-add-group")
+        (mc.handlings[0].request.headers.findAll("x-roles").toString()).contains("not-admin")
+        (mc.handlings[0].request.headers.findAll("x-roles").toString()).contains("compute:default")
+        (mc.handlings[0].request.headers.findAll("x-roles").toString()).contains("object-store:default")
+        (mc.handlings[0].request.headers.findAll("x-roles").toString()).contains("test-add-role")
+        (mc.handlings[0].request.headers.findAll("x-pp-user").toString()).contains("repose-test-add-user")
+        (mc.handlings[0].request.headers.findAll("x-pp-user").toString()).contains("username")
+    }
+
+    // REP-2464: Auth filter should add headers not replace headers
+    @Ignore ("We can turn on when impersonator role to header merge in to branch")
```

Comment:
Not that I know of... I would think that should be of REP-2607? Not sure why 2464 is annotated here.